src/Platforms/MariaDBPlatform.php

```diff
@@ -52,4 +112,50 @@ protected function initializeDoctrineTypeMappings(): void
 
         $this->doctrineTypeMapping['json'] = Types::JSON;
     }
+
+    /**
+     * Generate SQL snippets to inverse the aliasing of JSON to LONGTEXT for applicable MariaDb versions and platforms.
+     *
+     * MariaDb aliases columns specified as JSON to LONGTEXT and sets a CHECK constraint to ensure the column
+     * is valid json. This function generates the SQL snippets which inverse this aliasing i.e. report a column
+     * as JSON where it was originally specified as such instead of LONGTEXT.
+     *
+     * The CHECK constraints are stored in information_schema.CHECK_CONSTRAINTS.
+     *
+     * Applies only for MariaDBPlatforms that set json columns as JSON.
+     *
+     * Uses executable comments to ensure the inversing happens on versions of MariaDb which set the column check,
+     * namely 10.4.3 onwards.
+     *
+     * @return array{string, string}
+     */
+    public function getColumnTypeSQLSnippets(string $tableAlias = 'c'): array
```

Comment:
Shouldn't this override happen in `MariaDb1043Platform` instead?

Reply:
I put it in the base MariaDbPlatform to allow flexibility to choose between JSON and LONGTEXT columns for json types. It calls the parent where LONGTEXT is chosen and the executable comments prevent the additional SQL running on versions of MariaDb that don't support it. In other words, I saw adding the method to the base platform as a flexible way of supporting json columns on all versions of MariaDb. If it's confusing, I can move it to the specific platform (i.e. 1043).

Reply:
Yes, let's move it. This also allows us to get rid of those conditional comments.

Reply:
Will you have time to make this adjustment?

Reply:
Will do. I may have time tomorrow but more likely at the weekend.

Reply:
I have updated the pull request as follows:

1. I've moved the `getColumnTypeSQLSnippets()` method to `MariaDb1043Platform`. Since it made little sense to override certain methods in `MariaDBPlatform` and others in `MariaDb1043Platform`, I also moved `getListTableColumnsSQL()` and `getColumnTypeSQLSnippets()` meaning there are no changes to the base `MariaDBPlatform` class.

2. I added a couple of tests to ensure MariaDb's treatment of collation and character set did not cause problems (MariaDb forces collation to binary and reports it to doctrine so without these patches, the comparator sees a change from the general collation it was expecting and was generating incorrect migrations).

3. I spent time tracking through where this enforced collation triggered the unnecessary migrations and, as a result, moved the fix from the originally private `columnToArray()` method to `getColumnDeclarationSQL()`, the column parameter for which was generated by `columnToArray()`. This is the method called by `columnsEqual()` to detect table changes. The amended patch is functionally equivalent to the earlier version. I still so no way of moving this into the schema manager as columnsEqual() is called on the metadata based table and the introspected table and the schema manager doesn't see the metadata based table. 

4. Once the collation issues are dealt with, contrary to what I'd said earlier, no migration is generated when moving from the old to new platforms, i.e. existing json tables are not upgraded to native JSON. This is because the columns are marked with DC2Type comments which override detection of underlying column types i.e. old and new columns appear equal to the comparator. I have added a final patch (it currently has effect only for MariaDb1043 platforms) which also compares the reported database data type where DC2Types are specified. This enables the comparator to detect the LONGTEXT to JSON change. If similar issues occur in future, the functionality could be activated for other platforms by overriding the new `AbstractPlatform->columnDeclarationsMatch()` method but the relevant platform schema manager would also need patching to detect the schema change and I have only done this for `MySQLSchemaManager` in the proposed patch. 

I have kept the above separate to the original patches to make your review easier (at least, that's what I intended). If you're okay with the proposals, I'd be happy to squash items 1, 2 and 3 into the original patches but I think 4 makes sense kept separate or, perhaps, dropped and users would then either have to manually migrate MariaDb JSON columns or live with the old LONGTEXT declarations for old tables. 

Sorry this has taken me a while to amend - it turned out to be rather more involved than I at first thought and I wanted to make sure I had tested it locally before burdening you.